src/sentry/static/sentry/app/views/settings/organizationMembers/organizationMemberDetail.tsx

```diff
@@ -0,0 +1,421 @@
+import {RouteComponentProps} from 'react-router/lib/Router';
+import {browserHistory} from 'react-router';
+import React from 'react';
+import * as Sentry from '@sentry/browser';
+import styled from 'react-emotion';
+
+import {Member, Organization, Team} from 'app/types';
+import {Panel, PanelBody, PanelHeader, PanelItem} from 'app/components/panels';
+import {
+  addErrorMessage,
+  addLoadingMessage,
+  addSuccessMessage,
+} from 'app/actionCreators/indicator';
+import {inputStyles} from 'app/styles/input';
+import {removeAuthenticator} from 'app/actionCreators/account';
+import {resendMemberInvite, updateMember} from 'app/actionCreators/members';
+import {t, tct} from 'app/locale';
+import AsyncView from 'app/views/asyncView';
+import AutoSelectText from 'app/components/autoSelectText';
+import Button from 'app/components/button';
+import Confirm from 'app/components/confirm';
+import DateTime from 'app/components/dateTime';
+import ExternalLink from 'app/components/links/externalLink';
+import Field from 'app/views/settings/components/forms/field';
+import NotFound from 'app/components/errors/notFound';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import TeamSelect from 'app/views/settings/components/teamSelect';
+import Tooltip from 'app/components/tooltip';
+import recreateRoute from 'app/utils/recreateRoute';
+import space from 'app/styles/space';
+import withOrganization from 'app/utils/withOrganization';
+
+import RoleSelect from './inviteMember/roleSelect';
+
+const MULTIPLE_ORGS = t('Cannot be reset since user is in more than one organization');
+const NOT_ENROLLED = t('Not enrolled in two-factor authentication');
+const NO_PERMISSION = t('You do not have permission to perform this action');
+const TWO_FACTOR_REQUIRED = t(
+  'Cannot be reset since two-factor is required for this organization'
+);
+
+type Props = {
+  organization: Organization;
+} & AsyncView['props'] &
+  RouteComponentProps<
+    {
+      orgId: string;
+      memberId: string;
+    },
+    {}
+  >;
+
+type State = {
+  roleList: Member['roles'];
+  selectedRole: Member['role'];
+  member: Member | null;
+} & AsyncView['state'];
+
+class OrganizationMemberDetail extends AsyncView<Props, State> {
+  getDefaultState() {
```

Comment:
@billyvg any reason you aren't typing the return value here?

Reply:
Basically because it didn't error